DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Operations/Send/RootObjectBuilder.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using Speckle.Connectors.DUI.Bindings;
+using Speckle.Converters.Common;
+using Speckle.Core.Models;
+using System.Threading;
+using Autodesk.Revit.DB;
+using Speckle.Converters.RevitShared.Helpers;
+using Speckle.Connectors.Revit.HostApp;
+using System.Linq;
+
+namespace Speckle.Connectors.Revit.Operations.Send;
+
+public class RootObjectBuilder
+{
+  private readonly SendSelection _sendSelection;
```

Comment:
"having too many interfaces" <- there may be cases there we might not need the interface, but for many things, it is cheap to make the interface and have it, rather the build up a huge cost of having to change behavious and make interfaces for a type down the line, probably by that point multiple types.

Concrete types, bind services together - you cannot instantiate one service without another and this creates tight coupling. This tight coupling makes mocking and unit testing hard and makes it harder to replace one thing with another, following the Liskov subst. principal of SOLID. It also fulfils the O in SOLID, because it is open for expansion in this way and the I because it allows for inversion of control - if we have an ISendSelection then how it functions is opaque to the types that depend upon it, provided it returns the data required.

That said, I don't know what's in SendSelection without hunting it down, so it could be that in this case the concrete type isn't as important - I am a bit reluctant to accept that argument as this is being injected and so probably has dependencies of it's own or is hiding something else